### PR TITLE
#79: Creating Data Points For 3D Graph

### LINKED ISSUE #79: Creating Data Points For 3D Graph
# Required 
We need a data point component that takes on all information from a entry from the CSV file. Its coordinates for this iterations purpose can be anything for testing and interacting purposes but we need to find a way to define coordinate positions in relation to axis.  When this data point is selected a menu should pop up populated with the data of that specific point.  Have the size easily variable in case we want the size to be able to be adjusted if the graph itself is made bigger/smaller 
----
Dev member must keep in mind that this component must be used eventually with other components for the graph so keep compatibility in mind.

### PR DESCRIPTION
**Pull request by Subzeero**
Closes #79 

# What was the issue
One of our core requirements for our project requires a plot for analyzing data. This PR includes one of the necessary components for this plot: the data points. 

# What was done and why
I created two React components: DataPoint and DataPointMenu. The DataPoint component will represent each point of data that will be plotted. The DataPointMenu appears when a DataPoint is selected and it display information about the DataPoint. The selected point is shared using a PointSelectionContext which stores the selected DataPoint id number in its state.

# How it was tested
I used the WebXR emulator to test the requested functionality.

### CODE REVIEW

File: my-webxr-app/src/App.tsx [hunk 1]
```diff
@@ -4,29 +4,35 @@ import '@react-three/fiber'
...
+        <Canvas>
+          <XR>
+            <Sky sunPosition={[0, 1, 0]} />
+            <Floor />
+            <ambientLight />
+            <pointLight position={[10, 10, 10]} />
+            <Controllers />
+            {/*<Button position={[0, 1.5, -1]} />*/}
+            {/*<RotatingBox position={[0.8, 1.5, -1]} />*/}
+            {/*<RotatingBox position={[-0.8, 1.5, -1]} />*/}
```
Comment (Jesster2829) [inline my-webxr-app/src/App.tsx:27]: Can probably remove this commented out code

File: my-webxr-app/src/contexts/PointSelectionContext.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+export const usePointSelectionContext = () => {
+  const context = useContext(PointSelectionContext);
+  if (!context) {
+    throw new Error(
+      "You must use this context within a PointSelectionProvider!",
+    );
+  }
+  return context;
+};
```
Comment (Jesster2829) [inline my-webxr-app/src/contexts/PointSelectionContext.tsx:38]: Add some documentation for what is happening here

**Review by Jesster2829 — CHANGES_REQUESTED** [review]
Just add documentation otherwise good to go

**Comment by Subzeero** [comment]
Addressed the review concerns. Also fixed the minor issue when two controllers hovered over the same DataPoint and the hover effect was prematurely removed. Simply switched from a Boolean state to a count of hovers.

**Review by LongQuanNguyen — COMMENTED** [review]
The components folder is with 'C' capitalized. Should it be in lowercase?

File: my-webxr-app/src/components/DataPoint.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/DataPoint.tsx:1]: Needs unit tests

File: my-webxr-app/src/components/DataPointMenu.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/components/DataPointMenu.tsx:1]: needs unit tests

File: my-webxr-app/src/contexts/PointSelectionContext.tsx
Comment (matthew-buglass) [inline my-webxr-app/src/contexts/PointSelectionContext.tsx:1]: needs unit tests

**Review by matthew-buglass — CHANGES_REQUESTED** [review]
We just need to add unit tests for the components

File: my-webxr-app/src/contexts/PointSelectionContext.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  setSelectedDataPoint: React.Dispatch<
+    React.SetStateAction<PointSelectionContext["selectedDataPoint"]>
+  >;
+}
+
+/**
+ * Create the Context.
+ */
+export const PointSelectionContext =
+  createContext<PointSelectionContext | null>(null);
```
[the remark this answers is not in the document]
Comment (LongQuanNguyen) [reply my-webxr-app/src/contexts/PointSelectionContext.tsx:21]: Yes, Comment resolve during pairing

File: my-webxr-app/src/contexts/PointSelectionContext.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    </PointSelectionContext.Provider>
+  );
+};
+
+/**
+ * Provide a type-guaranteed context (not null) for use within components.
+ * Call this function instead of useContext(PointSelectionContext).
+ */
+export const usePointSelectionContext = () => {
+  const context = useContext(PointSelectionContext);
```
[the remark this answers is not in the document]
Comment (LongQuanNguyen) [reply my-webxr-app/src/contexts/PointSelectionContext.tsx:53]: The object is returned so it is guaranteed to be not null -- per discussion during paring

**Review by Jesster2829 — APPROVED** [review]
Pre approval before commit, have witnessed all changes in chat but will push after his assignment is done

**Pull request merged**